cita-executor/core/src/libexecutor/block.rs

```diff
@@ -217,43 +189,66 @@ impl ExecutedBlock {
+                    ExecutedException::VM(ExecErr::Evm(EVMError::InvalidOpcode)) => {
+                        Some(ReceiptError::BadInstruction)
+                    }
+                    ExecutedException::VM(ExecErr::Evm(EVMError::OutOfGas)) => {
+                        Some(ReceiptError::OutOfQuota)
+                    }
+                    ExecutedException::VM(ExecErr::Evm(EVMError::InvalidJumpDestination)) => {
+                        Some(ReceiptError::BadJumpDestination)
+                    }
+                    ExecutedException::VM(ExecErr::Evm(EVMError::OutOfStack)) => {
+                        Some(ReceiptError::OutOfStack)
+                    }
+
+                    ExecutedException::VM(ExecErr::Evm(EVMError::MutableCallInStaticContext)) => {
+                        Some(ReceiptError::MutableCallInStaticContext)
+                    }
+                    ExecutedException::VM(ExecErr::Evm(EVMError::OutOfBounds)) => {
+                        Some(ReceiptError::OutOfBounds)
+                    }
+
+                    ExecutedException::VM(ExecErr::Evm(EVMError::Internal(_))) => {
+                        Some(ReceiptError::ExecutionInternal)
+                    }
+                    ExecutedException::AuthError(AuthenticationError::NoTransactionPermission) => {
+                        Some(ReceiptError::NoTransactionPermission)
+                    }
+                    ExecutedException::AuthError(AuthenticationError::NoContractPermission) => {
                         Some(ReceiptError::NoContractPermission)
                     }
-                    ExecutionError::Authentication(AuthenticationError::NoCallPermission) => {
+                    ExecutedException::AuthError(AuthenticationError::NoCallPermission) => {
                         Some(ReceiptError::NoCallPermission)
                     }
-                    ExecutionError::Internal { .. } => Some(ReceiptError::ExecutionInternal),
-                    ExecutionError::TransactionMalformed { .. } => {
-                        Some(ReceiptError::TransactionMalformed)
-                    }
+                    ExecutedException::AuthError(AuthenticationError::TransactionMalformed {
+                        ..
+                    }) => Some(ReceiptError::TransactionMalformed),
                     _ => Some(ReceiptError::Internal),
                 };
 
                 let schedule = TxGasSchedule::default();
                 let sender = *t.sender();
-                let tx_quota_used = match err {
-                    ExecutionError::Internal(_) => t.gas,
-                    _ => cmp::min(
-                        self.state
-                            .borrow_mut()
-                            .balance(&sender)
-                            .unwrap_or_else(|_| U256::from(0)),
-                        U256::from(schedule.tx_gas),
-                    ),
+                let tx_quota_used = match receipt_error {
```

Comment:
Delete the `cmp::min` logic will lead incompatible, the logic is different as before.